ethereum/core/src/main/java/tech/pegasys/teku/core/ForkChoiceUtil.java

```diff
@@ -301,14 +292,20 @@ private static boolean isFinalizedAncestorOfJustified(
 
   private static Optional<BlockImportResult> checkOnBlockConditions(
       final BeaconBlock block,
-      final BeaconState preState,
+      final BeaconState blockSlotState,
       final ReadOnlyStore store,
       final ForkChoiceStrategy forkChoiceStrategy) {
     final UInt64 blockSlot = block.getSlot();
-    if (preState == null) {
+    if (blockSlotState == null) {
       return Optional.of(BlockImportResult.FAILED_UNKNOWN_PARENT);
     }
-    if (preState.getSlot().compareTo(blockSlot) >= 0) {
+    if (!blockSlotState.getSlot().equals(blockSlot)) {
+      return Optional.of(BlockImportResult.FAILED_INVALID_ANCESTRY);
+    }
+    if (blockSlot.isGreaterThan(UInt64.valueOf(GENESIS_SLOT))
+        && !get_block_root_at_slot(blockSlotState, blockSlot.minus(1))
+            .equals(block.getParentRoot())) {
+      // Block is at same slot as its parent or the parent root doesn't match the state
```

Comment:
Because the state has now already had slots processed up to the new block slot, the ancestry check is more complex.  We still need to ensure that `block.getSlot() > getBlock(block.getParentRoot()).getSlot()` but we don't have the actual parent block to get its slot.

So we check that:
1. the state we have is at the block slot (not ahead which indicates the block is invalid and not behind which suggests the caller didn't process empty slots
2. the block root at the slot before the block is the block's parent root. This ensures the block doesn't have the same slot as its parent.